core/common/libp2p/cbor_buffering.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common/libp2p/cbor_buffering.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(fc::common::libp2p, CborBuffering::Error, e) {
+  return "CborBuffering::Error";
+}
+
+namespace fc::common::libp2p {
+  using Head = CborBuffering::Head;
+
+  outcome::result<Head> Head::first(size_t &more, uint8_t first) {
+    auto type = Type{(first & 0xe0) >> 5};
+    switch (type) {
+      case Type::Unsigned:
+      case Type::Signed:
+      case Type::Bytes:
+      case Type::Text:
+      case Type::Array:
+      case Type::Map:
+      case Type::Tag:
+        break;
+      default:
+        return Error::INVALID_HEAD_TYPE;
+    }
+    Head head{type, 0};
+    first &= 0x1f;
+    if (first < 24) {
+      more = 0;
+      head.value = first;
+    } else {
+      switch (first) {
+        case 24:
```

Comment:
Please don't leave numbers as magic constants.
Comment with a link to the spec or some description would be useful.

It could be a single comment or multiple - each per constant if needed (e.g. different meaning of the underlying data).